test/Constraints/rdar60898369.swift

```diff
@@ -0,0 +1,12 @@
+// RUN: %target-typecheck-verify-swift -disable-availability-checking -swift-version 5
+// REQUIRES: objc_interop
+
+import Foundation
+
+func test(bookmark: Data, relativeURL: URL?, resolutionOptions: URL.BookmarkResolutionOptions) throws {
```

Comment:
Could we eliminate the Foundation and `objc_interop` dependencies here? This should reproduce with Swift 4.2 @availables on all platforms.

Reply:
That's what I thought but I couldn't reduce this unfortunately, here is what I tried:

```
struct S {
  @available(swift, obsoleted: 4.2)
  init(_: Int) throws {}

  @available(swift, introduced: 4.2)
  init?(_: Int) throws {}
}

_ = try S(42)
```

But it type-checks correctly even without this change. I'm not as proficient in all of the special cases related to availability checking to figure out what exactly I had to do...

Reply:
```swift
struct S {
  @available(swift, obsoleted: 4.2)
  init(foo: Int) throws {}

  @available(swift, introduced: 4.2)
  init?(foo: Int) throws {}
}

_ = try S(foo: 42)
```

This gives me `ambiguous use of 'init(foo:)'`. I guess the label is important?

Reply:
Interesting, I guess so...

Reply:
Thank you, @theblixguy! I have updated the test-case.